Clarification for `parent` and `ancestor` hierarchical relationships (#53855)
WordPress/gutenberg: docs/how-to-guides/block-tutorial/nested-blocks-inner-blocks.md

} );
```

> ## Using Hierarchical Context in InnerBlocks

Reviewer: I personally (maybe because I'm not a native English speaker) fund Hierarchical Context to be much more difficult to understand than Parent and Ancestor. I would love to better understand what the concerns / struggles are with that verbiage that are behind this change :)

Author: > find Hierarchical Context to be much more difficult to understand than Parent and Ancestor.

I agree. `Hierarchical Context” is clumsy. I’m mostly struggling with the previous “Child InnerBlocks”. That seem clumsy and the “Child” seems overqualifying. Maybe we just drop the “Child” from the heading and go with “InnerBlocks: Parents and Ancestors”?

> A common pattern for using InnerBlocks is to create a custom block that will be included _only_ in the InnerBlocks. This allows InnerBlocks to have a hierarchical context for nesting child blocks.

Reviewer: I feel like this is still somewhat convoluted / complex.

Maybe something like this would work better:

```suggestion
A common pattern in the editor are blocks that should only be available to users when they are nested inside specific other blocks. We have two mechanisms for that. A block can define its `parent` when the block needs to be a direct descendant of said block, or it can define an `ancestor` if it can appear anywhere nested within said block.
```

> ### Defining Parent Block Context

Reviewer: ```suggestion
### Restricting a block to only exist as a direct descendent of a specific block

When a block should only exist as a direct descendant of another block, we can use the `parent` block setting. When a `parent` has been added, the block can only be inserted as a direct child of the parent block.
```

> An example of this is the Column block, which is assigned the `parent` block setting. This allows the Column block to only be available in its parent Columns block context. Otherwise, the Column block will not be available as an option within the block inserter. See [Column code for reference](https://github.com/WordPress/gutenberg/tree/HEAD/packages/block-library/src/column).

Reviewer: ```suggestion
An example of this is the Column block, which is setting the `parent` block setting to `[ "core/columns" ]`. This allows the Column block to only be available as a direct descendant of the Columns block. Otherwise, the Column block will not be available within the block inserter. See [Column code for reference](https://github.com/WordPress/gutenberg/tree/HEAD/packages/block-library/src/column).
```

> When defining a direct child block, use the `parent` block setting to define which block is the parent. This prevents the child block from showing in the inserter outside of the InnerBlock it is defined for.

Reviewer: With the changes I suggested above I would say this paragraph would become redundant

}
```

> ### Defining Ancestor Block Context
> 
> Another example is using the `ancestor` block setting. The Comment Author Name block utilizes the `ancestor` setting to assign the Comment Template block to allow it to be a descendant of the Comment Template block while refraining from explicitly assigning it a direct child hierarchy. 

Reviewer: ```suggestion
### Restricting a block to only exist nested anywhere inside a specific block

When we have a block that should only exist nested anywhere inside a specific block we can use the `ancestors` block setting. When an `ancestor` has been added, the block can only be inserted inside an inner blocks area if the ancestor is present anywhere in the nested structure. But it doesn't need to be a direct parent. 

An example of using the `ancestor` block setting is the Comment Author Name block. It utilizes the `ancestor` setting to assign the Comment Template block. This means we can only use the Comment Author block inside the Comment Template. But we can for example wrap the Comment Author inside a Group as long as that Group lives inside the Comment Template.
```

> This context allows the Comment Author Name block to be anywhere in the hierarchical tree, and not _just_ a direct child of the parent Comment Template block, while still limiting its availability within the block inserter to only be visible an an option to insert if the Comment Template block is available.

Reviewer: I find this change a little more unclear than the original description. 

In the `parent` section above it starts with a generalized explanation first and then dives into an example. This section here is now missing that general section entirely. (It had that in the prior documentation). I would love to see a more general explanation before it dives into the example.
